compass/landice/mesh.py

```diff
@@ -167,6 +167,18 @@ def set_cell_width(self, section, thk, vx=None, vy=None,
         spacing = np.interp(lspd, [low_log_speed, high_log_speed],
                             [max_spac, min_spac], left=max_spac,
                             right=min_spac)
+
+        # Clean up where we have missing velocities. These are usually nans
+        # or the default netCDF _FillValue of ~10.e36
+        missing_data_mask = np.logical_or(
+                               np.logical_or(np.isnan(vx), np.isnan(vy)),
+                               np.logical_or(np.abs(vx) > 1.e5,
+                                             np.abs(vy) > 1.e5))
+        spacing[missing_data_mask] = max_spac
+        logger.info(f'Found {np.sum(missing_data_mask)} points in input '
+                    f'dataset with missing velocity values. Setting '
+                    f'velocity-based spacing to maximum value.')
```

Comment:
This does not seem like a general solution (what if the missing data was on an ice shelf or an ice stream?) but will work for this test case.  

The simplest thing I can think of to deal with the holes is the following (pseudocode):
```
do while still missing cells:
    loop over missing cells:
        if this cell is missing but has a nonmissing neighbor(s):
            assign it the average value of its nonmissing neighbors
```
As written, it would be dependent on the arbitrary order in which cells are looped over.  It also will probably be ugly if the hole is large.  But so far we've only seen small holes, and we don't care a lot about how we patch them - just that they are patched somehow.

The other way I can think to do it is to built an interpolation triangulation out of the good cells and use it to extrapolate into the missing cells.  Or build a kd tree and use that to take the value of the nearest good cell.
But I suspect either would be very computationally expensive relative to the previous idea, at least if we typically expect just a small holes.

Reply:
My thought here was that it would be better to use the data that are actually available. For example, the cell spacing set by distance to the grounding line or ice edge would over-ride the maximum distance set by the `use_speed` config option. But I can see how that would be problematic if the missing data were in the middle of the Ross Ice Shelf (i.e., fast flowing, but far from both grounding line and ice edge). The best solution might actually be case dependent and require some judgment call based on the region in question.
I think a better solution than a loop would just be to use `scipy.interpolate.NearestNDInterpolator`. I will give that a try.

Reply:
Would it be possible to patch the holes in the velocity data outside of compass and use the patched data set?  It seems like it would be worth doing a front propagation technique to fill in the holes.  I believe https://github.com/scikit-fmm/scikit-fmm provides such a capability but it likely wouldn't be fast enough to want to run it every time you make a mesh (especially for testing).

Reply:
This is what I was thinking of in particular: 

Final example here:
https://scikit-fmm.readthedocs.io/en/latest/#examples
Code here:
https://scikit-fmm.readthedocs.io/en/latest/examples.html#extension-velocities-example-py

Reply:
@trhille, your idea is also a good one.  I haven't had much luck with `scipy`'s interpolators in the past.  They're designed for interpolating a lot of points in a given field, rather than for interpolating a lot of fields at a given set of points, so they're not useful for remapping data.  But they might work fine for this particular case.  What I said still holds, though.  It would make more sense to do this once and have a filled-in velocity field that you use instead of the one with holes.

Reply:
My concern with doing this outside of compass is that it adds another separate step that makes the reproducible workflow harder to track. However, maybe we should have a step that checks for missing values and fills them like I have done here, and then print a warning that this might not be a universal solution and that mesh quality should be checked manually? On the other hand, few of us are likely to look through the case output log thoroughly enough to see that warning.

Reply:
I really think it's legit to have steps to preprocess the input data files that aren't part of the compass steps you run on a regular basis.  But if you can come up with a good way to fill in the gaps that doesn't make the test case take significantly longer, it's fine to keep it here.

We rerun the ocean mesh generation (at low resolution) as part of our regression testing.  I think you would want to do that in MALI, too (but that's up to you) so you presumably want a process that doesn't take more time than other regression tests if possible.

Reply:
Okay, so maybe we create a generalized tool for pre-processing data that lives in compass/landice. So it would still be part of compass, but it doesn't need to be called during mesh generation.

Reply:
Thanks @xylar. @matthewhoffman, what do you think? I feel like adding a pre-processing tool is beyond the scope of this PR. I can add some comments and/or a warning message that makes it clear that this isn't a universal fix.

Reply:
And I can try to get another PR with that pre-processing tool opened in time for use in PR #324.